### ISSUE TITLE
Change behaviour to exit with non-zero status code when no matching files

### ISSUE THREAD

**Issue by AndreaCrotti** [stylelint/stylelint#3880]
with version

``` $ stylelint --version
9.9.0
```

running something like

`$ stylelint /path/that/doesnt/exist`

happily return 0 and doesn't complain.
That's a very strange behaviour for me because it doesn't seem too useful to pass a directory that doesn't exist.

**Comment by jeddy3** [comment]
@AndreaCrotti Thanks for the report.

This is the intended behaviour since `8.0.0` (see https://github.com/stylelint/stylelint/pull/2464) as:
- it matched ESLint's behaviour at the time
- avoided adding complexity to the `ignore` functionality code

However, it looks like ESLint's behaviour has since changed:

```console
npx eslint /apth/fdg
No files matching '/apth/fdg' were found.
```

We should probably follow suit as it's arguably a better developer experience.

This will be a breaking change, and we'll need to reinstate the [`allowEmptyInput` option](https://github.com/stylelint/stylelint/pull/2117) that allowed users to opt-out of this behaviour. I'll label this ticket as "help wanted". @AndreaCrotti Feel free to contribute a PR making this change.

**Comment by AndreaCrotti** [comment]
Well I'm not sure you would ever want to check a directory that doesn't exist, would anyway really need that `allowEmptyInput` flag?
Anyway I'd love to contribute but I know almost nothing about JS/node so I might be a bit useless..

**Comment by jeddy3** [comment]
> Well I'm not sure you would ever want to check a directory that doesn't exist, would anyway really need that `allowEmptyInput` flag?

I believe we should error if _the glob_ doesn't match any files. If so, the use case outlined in the original `allowEmptyInput` issue, https://github.com/stylelint/stylelint/issues/2048, is still valid.

> Anyway I'd love to contribute but I know almost nothing about JS/node so I might be a bit useless..

Feel free to poke around. We all have to start somewhere with a new language. You can fork the repo, run `npm install` to install the dependencies, and then run `npm run watch` to start the interactive testing prompt. As you change files you'll see what breaks, and that can be a good learning experience in itself.
